### PR TITLE
Update deprecation policy to use new decorators

### LINKED ISSUE #1709: Update deprecation policy and explain deprecation decorators
## Does existing documentation need to be updated with new content?

- Which files need to be updated?

https://qiskit.org/documentation/deprecation_policy.html

- What content needs to be updated or replaced? (You can copy and paste)

At least https://qiskit.org/documentation/deprecation_policy.html#issuing-deprecation-warnings needs a rewriting to adapt it to the new @Eric-Arellano 's way https://github.com/Qiskit/qiskit-terra/pull/9676

### PR DESCRIPTION
**Pull request by Eric-Arellano**
### Summary

Closes https://github.com/Qiskit/qiskit-metapackage/issues/1709.

I didn't explain every single detail of the decorators because it is too subject to bit-rot. I figure it's better to have people read the original documentation.

### CODE REVIEW

**Comment by mtreinish** [comment]
This will need to wait until the Qiskit release right? Or are we ok documenting this on the published docs before the new decorators are included in a release?

**Comment by Eric-Arellano** [comment]
> This will need to wait until the Qiskit release right? Or are we ok documenting this on the published docs before the new decorators are included in a release?

I'm fine with waiting until the release to hit merge.

I mostly wanted to get this up for review while this deprecation stuff is all fresh in my head.

File: docs/deprecation_policy.rst [hunk 1]
```diff
@@ -123,11 +123,38 @@ change:
...
 ============================
 
-The proper way to raise a deprecation warning is to use the ``warn`` function
+The proper way to raise a deprecation warning is to use the decorators ``@deprecate_arg`` and
+``@deprecate_func`` from ``qiskit.util.deprecation`. These will generate a standardized message and
+and add the deprecation to that function's docstring so that it shows up in the docs.
+
+.. code-block:: python
+
+    from qiskit.util.deprecation import deprecate_arg, deprecate_func
```
Comment (mtreinish) [inline docs/deprecation_policy.rst]: I think this is supposed to be:
```suggestion
    from qiskit.utils.deprecation import deprecate_arg, deprecate_func
```

**Comment by Eric-Arellano** [comment]
Terra 0.24 is released, so this is now ready.

File: docs/deprecation_policy.rst [hunk 2]
```diff
@@ -123,22 +123,49 @@ change:
 Issuing deprecation warnings
 ============================
 
-The proper way to raise a deprecation warning is to use the ``warn`` function
+The proper way to raise a deprecation warning is to use the decorators ``@deprecate_arg`` and
+``@deprecate_func`` from ``qiskit.utils.deprecation``. These will generate a standardized message and
```
Comment (1ucian0) [inline docs/deprecation_policy.rst:127]: Is it possible to link to the docs directly? 
```suggestion
The proper way to raise a deprecation warning is to use the decorators :func:`~qiskit.utlis.deprecate_arg` and :func:`~qiskit.utlis.deprecate_func`. These will generate a standardized message and
```

File: docs/deprecation_policy.rst [hunk 3]
```diff
@@ -123,22 +123,49 @@ change:
...
+    def another_func(bad_arg: str, new_name: str):
+        pass
+
+Usually, you should set ``additional_msg: str `` with the format ``"Instead, use ..."`` so that
+people know how to migrate. Read those functions' docstrings for additional arguments like
+``pending: bool`` and ``predicate``.
+
+If you are deprecating in a non-Terra repo, set ``package_name``, e.g. to ``qiskit-nature``.
+Alternatively, if you prefer to use your own decorator helpers, then have them call
+``add_deprecation_to_docstring`` from ``qiskit.utils.deprecation``.
```
Comment (1ucian0) [inline docs/deprecation_policy.rst:148]: Same where. Is it possible to link to the doc?

**Comment by Eric-Arellano** [comment]
@1ucian0 we will soon move this document to Terra. We could set up intersphinx now, but it will be much easier if we instead land this as is. Then use cross references when migrating to Terra.

**Review by 1ucian0 — APPROVED** [review]

**Pull request merged**